sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -61,7 +61,8 @@ AZ_NODISCARD az_result az_keyvault_keys_client_init(
   AZ_CONTRACT_ARG_NOT_NULL(self);
   AZ_CONTRACT_ARG_NOT_NULL(options);
 
-  _az_credential_vtbl * const cred = (_az_credential_vtbl *)credential;
+  _az_credential * const cred = (_az_credential *)credential;
+  cred->_internal.http_client = &self->_internal.options._internal.http_client;
```

Comment:
Of course, the key_vault_key_client_options structure also needs to change the http_client to a transport_policy_options struct as well.